### PR TITLE
Applying NR 1 to CalculatorVector.h

### PR DESCRIPTION
**Pull request by Nicholas-Baron**
## Fixes #.
Applied [NR 1](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rnr-top) from the ISO C++ Guidelines to CalculatorVector.h

### Description of the changes:
- Move towards understandable and maintainable code

### CODE REVIEW

File: src/CalcManager/CalculatorVector.h [hunk 1]
```diff
@@ -141,7 +135,7 @@ class CalculatorVector
             }
 
             std::wstring expressionSuffix{};
-            hr = GetExpressionSuffix(&expressionSuffix);
+            hr = GetExpressionSuffix(&expressionSuffix); // Will always return S_OK
```
Comment (rudyhuyn) [inline src/CalcManager/CalculatorVector.h]: I would suggest to not add this comment, there is a risk this comment will become obsolete if the implementation of `GetExpressionSuffix` changes and will be misleading for developers.

**Comment by HowardWolosky** [comment]
@Nicholas-Baron  -- This was submitted as a draft.  Are you intending on marking this as "Ready for review", or do you just want to close it out?

File: src/CalcManager/CalculatorVector.h [hunk 2]
```diff
@@ -120,12 +114,12 @@ class CalculatorVector
 
     ResultCode GetString(_Out_ std::wstring* expression)
     {
-        ResultCode hr = S_OK;
         unsigned int nTokens = 0;
-        std::pair<std::wstring, int> currentPair;
-        hr = this->GetSize(&nTokens);
+        auto hr = this->GetSize(&nTokens);
```
Comment (HowardWolosky) [inline src/CalcManager/CalculatorVector.h]:
```suggestion
        ResultCode hr = this->GetSize(&nTokens);
```

I don't see any benefit to using `auto` in this scenario.

**Comment by HowardWolosky** [comment]
Can you please also update your PR description with a link to `NR 1 from the ISO C++ Guidelines`? thanks.

**Comment by Nicholas-Baron** [comment]
@HowardWolosky The initial description has the link to [NR 1](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rnr-top).

**Pull request merged**